shared/components/ember-output.jsx

```diff
@@ -0,0 +1,90 @@
+/* global React */
+
+export default React.createClass({
+    displayName: 'EmberOutput',
+
+    propTypes: {
+        source  : React.PropTypes.string.isRequired,
+        context : React.PropTypes.object.isRequired
+    },
+
+    injectMessages: function (locale, messages) {
+        var container  = this.container;
+        var lookupName = locale.toLowerCase();
+
+        if (container.has('locale:' + lookupName)) {
+            container.unregister('locale:' + lookupName);
+        }
+
+        container.register('locale:' + lookupName, {
+            locale:   locale,
+            messages: messages
+        }, {
+            instantiate: false
+        });
+    },
+
+    componentWillReceiveProps: function (nextProps) {
+        if (nextProps.source !== this.props.source) {
+            this.setState({});
+        }
+
+        var locale = Ember.get(nextProps, 'locales');
+
+        if (typeof locale === 'string') {
+            this.injectMessages(locale, nextProps.messages);
+            this.app.intl.set('locales', Ember.makeArray(locale));
+        }
+
+        this.controller.setProperties(nextProps);
+    },
+
+    componentWillUnmount: function () {
+        var app = this.app;
+
+        if (!app.isDestroying && !app.isDestroyed) {
+            app.destroy();
+        }
+    },
+
+    componentDidMount: function () {
+        var domElement = this.refs.placeholder.getDOMNode();
+        var locales    = Ember.makeArray(this.props.locales);
+
+        this.app = Ember.Application.extend().create({
+            rootElement: domElement,
+            ready: function () {
+                this.intl.setProperties({
+                    locales:        locales,
+                    defaultLocales: ['en-US']
```

Comment:
I'm torn if removing it is a good idea or not from ember-intl.  It's not used here so I can safely remove it since all the message objects per-locale has the messages.  

But in the case where you supply a message key and the current locale doesn't have that message, I would think it should fallback to a default locale before throwing an error.